### DISCUSSION TITLE
use fetch in +layout.server.ts load function without including the fetched url in client side code

### DISCUSSION THREAD

**Discussion by Nikhil1920** [sveltejs/kit#9384] [Q&A] [answered]
I am fetching data from a third party api using the load function's fetch in +layout.server.ts and using that data in +page.server.ts 

But the url being fetched is included in the uses dependencies array in the response html even though it is never being called in client side

**Comment by jeblister** [comment]
The problem is likely due to the server-side rendering (SSR) that SvelteKit provides. When the page is rendered on the server, the code in +page.server.ts is executed and the data from the API is included in the response HTML. 

When the page is rendered in the browser, the data from the API is not fetched again and the URL will still be included in the HTML. 

To prevent this, you can configure the page to not be rendered on the server with the SSR option in the page options.

**Comment by geoffrich** [comment]
Can you provide a repro? I spot-checked my site that fetches data from an external API in `+page.server.ts` and the API's URL does not show in the source of the page.

**Comment by Nikhil1920** [comment]
The issue is occurring when data is being fetched from an external api in `+layout.server.ts` you can find the repo [here](https://github.com/Nikhil1920/fetch-url-demo/tree/master/src/routes/fetch-demo)

A live version is deployed [here](https://joyful-jalebi-39709c.netlify.app/fetch-demo)

if you go to view sources on that page you can find the jsonplaceholder url from which we are fetching the data

**Reply by Nikhil1920** [reply]
@geoffrich

**Comment by Nikhil1920** [comment, marked as answer]
this was fixed in [1.18.0 release](https://github.com/sveltejs/kit/releases/tag/%40sveltejs%2Fkit%401.18.0) thanks for the help
